### PR TITLE
UpperBound Lints: Common Name and Locality Name

### PR DESCRIPTION
**Pull request by jddicki2**
Added upper bound lints relevant to common name and locality name.

### CODE REVIEW

File: lints/lint_ub_common_name_invalid.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+		return ResultStruct{Result: Pass}, nil
+	} else {
+		return ResultStruct{Result: Error}, nil
+	}
+}
+
+func init() {
+	RegisterLint(&Lint{
+		Name:          "e_ub_common_name_invalid",
+		Description:   "The 'Common Name' field must be less than 64 integers long.",
```
Comment (zakird) [inline lints/lint_ub_common_name_invalid.go]: This should explain _which_ common name field. A certificate can have multiple.

File: lints/lint_ub_common_name_invalid_test.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+package lints
+
+import (
+	"testing"
+)
+
+func TestUbCommonNameGood(t *testing.T) {
+	inputPath := "../testlint/testCerts/ubCommonNameGood.pem"
+	desEnum := Pass
+	out, _ := Lints["e_ub_common_name_invalid"].ExecuteTest(ReadCertificate(inputPath))
```
Comment (zakird) [inline lints/lint_ub_common_name_invalid_test.go]: what is `ub` here?

Comment (zakird) [inline lints/lint_ub_common_name_invalid_test.go]: Seems like naming "max_length" would be a lot more descriptive.

Comment (kumarde) [inline lints/lint_ub_common_name_invalid_test.go]: yeah, it was UB (upper bound) in the RFC

File: lints/lint_ub_common_name_invalid.go (hunk 1 above)
Comment (kumarde) [inline lints/lint_ub_common_name_invalid.go]: So this is issuer vs. subject? Are there others?

Comment (zakird) [inline lints/lint_ub_common_name_invalid.go]: those are the two I know of, but god only knows. This _is_ x509.

Comment (zakird) [inline lints/lint_ub_common_name_invalid.go]: OH OH OH OH!!! sub alt name can have DNs!!!!

Comment (zakird) [inline lints/lint_ub_common_name_invalid.go]: And also name constraints.

Comment (kumarde) [inline lints/lint_ub_common_name_invalid.go]: AHHHH

Comment (kumarde) [inline lints/lint_ub_common_name_invalid.go]: @jddicki2 for now, make this explicit. We'll need to figure out where all these fields apply carefully in future upper bound tests and make sure we're not missing anything

**Comment by zakird** [comment]
I might suggest also not measuring length in count of integer ;)

**Comment by kumarde** [comment]
@jddicki2 when you fix these changes, please also sync the fork with the master branch.

**Comment by jddicki2** [comment]
Sending a new PR with fixes soon

**Pull request closed without merging**